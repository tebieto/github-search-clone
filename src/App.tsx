import React, { lazy, Suspense } from 'react';
import { Redirect, Route, Switch } from 'react-router';
import { BrowserRouter } from 'react-router-dom';
import './App.css';
import ErrorBoundary from './components/ErrorBoundary';
import Loader from './components/Loader';
import { ACCESS_TOKEN_KEY } from './utils/constants';
const Login = lazy(() => import('./pages/Login'));
const Search = lazy(() => import('./pages/Search'));
const Result = lazy(() => import('./pages/Result'));
function App(): JSX.Element {
  const isLoggedIn = localStorage.getItem(ACCESS_TOKEN_KEY);
  return (
    <BrowserRouter>
      <Switch>
        <ErrorBoundary>
          <Suspense fallback={<Loader />}>
            <Route
              path={'/'}
              exact
              render={() =>
                isLoggedIn ? <Redirect to="/search" /> : <Login />
              }
            />
            <Route
              path={'/search'}
              exact
              render={() => (isLoggedIn ? <Search /> : <Redirect to="/" />)}
            />
            <Route
              path={'/search/results'}
              exact
              render={() => (isLoggedIn ? <Result /> : <Redirect to="/" />)}
            />
          </Suspense>
        </ErrorBoundary>
      </Switch>
    </BrowserRouter>
  );
}

export default App;
