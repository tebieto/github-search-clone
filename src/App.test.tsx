import React from 'react';
import App from './App';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { act } from 'react-dom/test-utils';

jest.mock('./utils/auth');

const waitForComponentToPaint = async (wrapper: { update: { (): void } }) => {
  await act(async () => {
    await new Promise((resolve) => setTimeout(resolve));
    wrapper.update();
  });
};

describe('App', () => {
  it('renders correctly', () => {
    const tree = mount(<App />);
    waitForComponentToPaint(tree);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
