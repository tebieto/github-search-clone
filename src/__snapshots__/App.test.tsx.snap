// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<BrowserRouter>
  <Switch>
    <ErrorBoundary>
      <Suspense
        fallback={<Loader />}
      >
        <Route
          exact={true}
          path="/"
          render={[Function]}
        />
        <Route
          exact={true}
          path="/search"
          render={[Function]}
        />
        <Route
          exact={true}
          path="/search/results"
          render={[Function]}
        />
      </Suspense>
    </ErrorBoundary>
  </Switch>
</BrowserRouter>
`;
